workitem/workitem_repository.go

```diff
@@ -230,15 +250,17 @@ func (r *GormWorkItemRepository) Reorder(ctx context.Context, position *app.Work
 		if err != nil {
 			return nil, errors.NewNotFoundError("work item", position.ID)
 		}
-		if aboveItemId == nil {
+		if *aboveItemId == "0" {
 			// Item is placed at last position
 			order = *belowItemOrder + float64(orderValue)
-		} else if *aboveItemId == res.ID {
+		} else if *aboveItemId == string(res.ID) {
```

Comment:
I think this can never be `true` at least for how you intended it to work. Check this example:

```go
package main

import (
	"fmt"
)

func main() {
	zero := "0"
	t := string(0)
	fmt.Printf("zero = '%+v'\nt = '%+v'\n", zero, t)
        // Outputs:
        // zero = '0'
        // t = ''
}
```

You should use [`FormatUint`](https://golang.org/pkg/strconv/#FormatUint) instead to convert an unsigned integer into a string.